spyder/plugins/application/widgets/install.py

```diff
@@ -0,0 +1,258 @@
+INSTALLING = _("Installing")
+FINISHED = _("Installation finished")
+PENDING = _("Pending update")
+CHECKING = _("Checking for updates")
+CANCELLED = _("Cancelled")
+
+
+class UpdateInstallation(QWidget):
+    """Update progress installation widget."""
+
+    def __init__(self, parent):
+        super(UpdateInstallation, self).__init__(parent)
+
+        # Left side
+        action_layout = QVBoxLayout()
+        progress_layout = QHBoxLayout()
+        self._progress_widget = QWidget(self)
+        self._progress_widget.setFixedHeight(50)
+        self._progress_bar = QProgressBar(self)
+        self._progress_bar.setFixedWidth(180)
+        self.cancel_button = QPushButton()
+        self.cancel_button.setIcon(ima.icon('DialogCloseButton'))
+        progress_layout.addWidget(self._progress_bar, alignment=Qt.AlignLeft)
+        progress_layout.addWidget(self.cancel_button)
+        self._progress_widget.setLayout(progress_layout)
+
+        self._progress_label = QLabel(_('Downloading'))
+
+        self.install_info = QLabel(
+            _("Downloading the latest Spyder version <br>"))
+
+        button_layout = QHBoxLayout()
+        self.ok_button = QPushButton(_('OK'))
+        button_layout.addStretch()
+        button_layout.addWidget(self.ok_button)
+        button_layout.addStretch()
+        action_layout.addStretch()
+        action_layout.addWidget(self._progress_label)
+        action_layout.addWidget(self._progress_widget)
+        action_layout.addWidget(self.install_info)
+        action_layout.addSpacing(10)
+        action_layout.addLayout(button_layout)
+        action_layout.addStretch()
+
+        # Layout
+        general_layout = QHBoxLayout()
+        general_layout.addLayout(action_layout)
+
+        self.setLayout(general_layout)
+
+    def update_installation_status(self, status):
+        """Update installation status (downloading, installing, finished)."""
+        self._progress_label.setText(status)
+        self.install_info.setText(status + _(" the latest version of Spyder."))
+        if status == INSTALLING:
+            self._progress_bar.setRange(0, 0)
+            self.cancel_button.hide()
+
+    def update_installation_progress(self, current_value, total):
+        """Update installation progress bar."""
+        self._progress_bar.setMaximum(total)
+        self._progress_bar.setValue(current_value)
+
+
+class UpdateInstallerDialog(QDialog):
+    """Spyder installer."""
+
+    # Signal to get the download progress
+    # int: Download progress
+    # int: Total download size
+    sig_download_progress = Signal(int, int)
+
+    # Signals
+    # Signal to get the current status of the update installation
+    # str: Status string
+    sig_installation_status = Signal(str)
+
+    def __init__(self, parent):
+
+        self.cancelled = False
+        self.status = NO_STATUS
+        self.thread_install_update = None
+        super(UpdateInstallerDialog, self).__init__(parent)
```

Comment:
```suggestion
        super().__init__(parent)
```

That was the old way of using `__init__`.